instagram/src/authenticate/authenticate.js

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import PostsPage from "../components/PostContainer/PostsPage";
+
+const authenticate = App => Login => {
+  return class extends React.Component {
+    constructor() {
+      super();
+      this.state = { loggedIn: false };
+    }
+
+    componentDidMount() {
+      if (localStorage.getItem("username")) {
+        this.setState({ loggedIn: true });
+      }
+    }
+
+    render() {
+      return this.state.loggedIn ? <PostsPage /> : <Login />;
```

Comment:
This works just as well as the method Dustin showed during his demo Wednesday, but try to implement it his way as well -- essentially it just forgoes the need to use a CDM hook or the need to place `loggedIn` on state and saves you about 10 lines of code, looking a lot cleaner